.github/workflows/goreleaser.yml

```
@@ -0,0 +1,47 @@
+name: goreleaser
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
```

Maybe consider only tagging version branches?
```suggestion
      - 'v[0-9].[0-9]+'
```

Not that I see other tags in the repo, but it would be weird to create a random one and have a release binary built for it. YMMV.

---

`goreleaser` will actually fail if the release tag name is not a semantic version. Will that be enough?

---

> `goreleaser` will actually fail if the release tag name is not a semantic version. Will that be enough?

Do you mean "fail" or just "skip"? If it skips, that seems fine. If it fails, would that cause CI runs to fail on any commit tagged with a non-version-shaped thing?

This is probably not that big a deal, it was just something I noticed in passing.

---

Can we call it `v*` and tag eg `v0.1.0`?

---

@creachadair the latter. Example failed run: https://github.com/hu13/gauth/runs/5268255400?check_suite_focus=true

@pcarrier yes!

---

There are already `v1.0` and `v1.1` tags, so you might want to avoid going back to v0. That said: it looks like the absence of a patch number keeps godoc.org from picking them up (the sumbd seems to have a commit from the 1st). So maybe that doesn't matter.